### PR TITLE
tests: move default functions out of utils

### PR DESCRIPTION
**Pull request by codingben**
### What this PR does

Move default functions from functional
tests to unit tests. Unit tests were using
utils of functional tests, where it should
not use them and use itself tests only.

### Release note

```release-note
None
```

### CODE REVIEW

**Review by enp0s3 — CHANGES_REQUESTED** [review]
@codingben Thank you for the initiative!
I would recommend to call the package `testing` and put it in  the following file: `pkg/virt-operator/components/testing/defaults.go` and these functions there. Then import it from the relevant testing files.

**Comment by codingben** [comment]
@dhiller @enp0s3 Got it, thanks! sounds good :)

**Comment by codingben** [comment]
@dhiller @enp0s3 Done, can you please take a look?

File: pkg/virt-operator/components/testing/defaults.go [hunk 1]
```diff
@@ -0,0 +1,82 @@
+package testing
```
Comment (EdDev) [inline pkg/virt-operator/components/testing/defaults.go:1]: Please add the header license.

File: pkg/virt-operator/kubevirt_test.go [hunk 2]
```diff
@@ -75,6 +73,7 @@ import (
 	kubecontroller "kubevirt.io/kubevirt/pkg/controller"
 	"kubevirt.io/kubevirt/pkg/monitoring/rules"
 	"kubevirt.io/kubevirt/pkg/testutils"
+	defaults "kubevirt.io/kubevirt/pkg/virt-operator/components/testing"
```
Comment (EdDev) [inline pkg/virt-operator/kubevirt_test.go:76]: It looks like another test helper was used here and place inside the production code imports.

I suggest we do not add this is bad grouping and place this import in its own group.

Also, it makes little sense to me that the name was aliased it to `defaults`. Could you explain the reason?

File: pkg/virt-operator/resource/apply/apps_test.go [hunk 3]
```diff
@@ -53,6 +51,7 @@ import (
 	"kubevirt.io/client-go/kubecli"
 
 	"kubevirt.io/kubevirt/pkg/controller"
+	defaults "kubevirt.io/kubevirt/pkg/virt-operator/components/testing"
```
Comment (EdDev) [inline pkg/virt-operator/resource/apply/apps_test.go:54]: Same

File: pkg/virt-operator/resource/apply/pdb_test.go [hunk 4]
```diff
@@ -20,6 +18,7 @@ import (
 	v1 "kubevirt.io/api/core/v1"
 	"kubevirt.io/client-go/kubecli"
 
+	defaults "kubevirt.io/kubevirt/pkg/virt-operator/components/testing"
```
Comment (EdDev) [inline pkg/virt-operator/resource/apply/pdb_test.go:21]: Same

**Review by EdDev — CHANGES_REQUESTED** [review]
The PR/commit is missing the reasoning for the change.
While in many other cases removing code from the big code collection in `tests/utils.go` is understood already, this scenario is different.

As I see it, the most important reason this is done is to drop the dependency from the unit-tests to the e2e tests.

The second problem this change encounters is the need to share unit tests helpers between different packages.
Usually this means that something really bad is happening with the tests, but if we try to ignore it and go to a solution in the form of a shared package for testing, we better explain it well and reason for it in the change (PR & commit text).

I am a bit unclear what this `components` package means (sound like another name for `utils` to me) and how the verb `testing` fits as a package name (usually it is a noun). If you explain it in the text it may help understand the logic.

If I need to pick a simple solution to this, I would just add a `libtest` package under `pkg/virt-operator` and name the file in it with `_test` postfix to be explicit about it containing test helpers. The package name will not have that postfix.

**Comment by enp0s3** [comment]
@codingben I was confused, for some reason I was thinking `components` directory already exists. I would go with @EdDev approach

File: pkg/virt-operator/components/testing/defaults.go (hunk 1 above)
Comment (codingben) [reply pkg/virt-operator/components/testing/defaults.go:1]: Done.

File: pkg/virt-operator/kubevirt_test.go (hunk 2 above)
Comment (codingben) [reply pkg/virt-operator/kubevirt_test.go:76]: I removed this package, and added the defaults functions to the test itself.

File: pkg/virt-operator/resource/apply/apps_test.go (hunk 3 above)
Comment (codingben) [reply pkg/virt-operator/resource/apply/apps_test.go:54]: Done.

File: pkg/virt-operator/resource/apply/pdb_test.go (hunk 4 above)
Comment (codingben) [reply pkg/virt-operator/resource/apply/pdb_test.go:21]: Done.

**Comment by codingben** [comment]
Let's not create a new package and just use the default functions in the test itself, it looks better and simpler now.

**Review by enp0s3 — COMMENTED** [review]
/approve

I would take this change.
As a follow-up PR I would refactor the `components` package. For me it will make sense to add product metadata into the `config` struct, and pass just the conifg as an argument. The current amount of arguments passed to these functions is a boiler-plate.

**Review by EdDev — APPROVED** [review]
Thanks!

**Pull request merged**